### PR TITLE
feat: add support for `yield return`

### PR DESCRIPTION
**Pull request by alangpierce**
Progress toward https://github.com/decaffeinate/decaffeinate/issues/688

### CODE REVIEW

**Comment by alangpierce** [comment]
Heh, sure. Yeah, I guess at this point it shouldn't be so bad. :-)

**Comment by alangpierce** [comment]
(I just updated the PR with mappers for `yield` and `yield from`.)

File: src/mappers/mapOp.ts [hunk 1]
```diff
@@ -48,3 +54,29 @@ function mapMultiplyOp(context: ParseContext, node: CoffeeOp) {
     mapAny(context, node.second)
   );
 }
+
+function mapYieldOp(context: ParseContext, node: CoffeeOp) {
```
Comment (eventualbuddha) [inline src/mappers/mapOp.ts]: Missing return type.

File: src/mappers/mapOp.ts [hunk 2]
```diff
@@ -48,3 +54,29 @@ function mapMultiplyOp(context: ParseContext, node: CoffeeOp) {
...
+    );
+  } else {
+    return new Yield(
+      line, column, start, end, raw, virtual,
+      mapAny(context, node.first)
+    );
+  }
+}
+
+function mapYieldFromOp(context: ParseContext, node: CoffeeOp) {
```
Comment (eventualbuddha) [inline src/mappers/mapOp.ts]: Missing return type.

File: src/mappers/mapOp.ts (hunk 1 above)
Comment (alangpierce) [inline src/mappers/mapOp.ts]: `mapMultiplyOp` was as well, and then I looked it up and learned that typescript can infer the type based on what's actually returned. But I guess you prefer to have explicit return types anyway? Would be nice if there was a tslint rule for that, although from a quick glance I don't see one.

**Pull request merged**